azext_edge/tests/edge/init/test_init_int.py

```diff
@@ -0,0 +1,167 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from enum import Enum
+from typing import Dict, List
+import pytest
+from os import mkdir
+
+from azext_edge.edge.providers.orchestration.common import KEYVAULT_ARC_EXTENSION_VERSION
+from ...helpers import run
+from ...generators import generate_names
+
+class ResourceKeys(Enum):
+    custom_location = "Microsoft.ExtendedLocation/customLocations"
+    mq = "Microsoft.IoTOperationsMQ/mq"
+    orchestrator = "Microsoft.IoTOperationsOrchestrator/targets"
+    connected_cluster = "Microsoft.Kubernetes/connectedClusters"
+
+
+@pytest.fixture(scope="function")
+def cluster_cleanup(cluster_setup, settings, tracked_files):
+    from ...settings import EnvironmentVariables
+    settings.add_to_config(EnvironmentVariables.rg.value)
+    settings.add_to_config(EnvironmentVariables.kv.value)
+    settings.add_to_config(EnvironmentVariables.cluster.value)
+    settings.add_to_config(EnvironmentVariables.sp_app_id.value)
+    settings.add_to_config(EnvironmentVariables.sp_object_id.value)
+
+    cert_dir = "certificates"
+    try:
+        mkdir(cert_dir)
+        tracked_files.append(cert_dir)
+    except FileExistsError:
+        pass
+    
+    yield {
+        "clusterName": settings.env.azext_edge_cluster,
+        "resourceGroup": settings.env.azext_edge_rg,
+        "keyVault": settings.env.azext_edge_kv,
+        "servicePrincipalAppId": settings.env.azext_edge_sp_app_id,
+        "servicePrincipalObjectId": settings.env.azext_edge_sp_object_id,
+        "CADirectory": cert_dir,
+    }
+    run(f"az iot ops delete --cluster {settings.env.azext_edge_cluster} -g {settings.env.azext_edge_rg} -y --no-progress --force")
+
+
+@pytest.mark.parametrize("scenario", [
+    pytest.param({"simulate_plc": True}, id="simualte plc"),
+    pytest.param({
+        "include_dp": True, "dp_instance": f"processor-{generate_names(max_length=5).lower()}"
+    }, id="data processor")
```

Comment:
btw with just these two scenarios, the run time is 32 minutes. 

I can break this up by having each scenario in a separate file (under an scenarios folder), each calling into a helper function that will run the tests. This should help with parallelization.

Reply:
as per meeting - will just have the params in the env variables